### PR TITLE
Unit tests for metrics_client.go

### PR DESCRIPTION
**Pull request by socaa**
@jszczepkowski 
@piosz

### CODE REVIEW

**Comment by jszczepkowski** [comment]
Very good PR! I had only some minor comments.

**Comment by socaa** [comment]
PTAL

@jszczepkowski 
@piosz

File: pkg/controller/podautoscaler/metrics/metrics_client_test.go [hunk 1]
```diff
@@ -19,136 +19,350 @@ package metrics
...
+		targetResource:        api.ResourceMemory,
+		desiredValue:          0,
+		reportedMetricsPoints: [][]metricPoint{},
+	}
+	tc.runTest(t)
+}
+
+func TestCPUEmptyMetricsForOnePod(t *testing.T) {
+	tc := testCase{
+		replicas:              4,
```
Comment (jszczepkowski) [inline pkg/controller/podautoscaler/metrics/metrics_client_test.go]: set it to 3

File: pkg/controller/podautoscaler/metrics/metrics_client_test.go [hunk 2]
```diff
@@ -19,136 +19,350 @@ package metrics
...
+		targetResource:        api.ResourceCPU,
+		desiredError:          fmt.Errorf("metrics obtained for 3/4 of pods"),
+		reportedMetricsPoints: [][]metricPoint{{metricPoint{}}, {{100, 1}}, {{400, 2}, {300, 3}}},
+	}
+	tc.runTest(t)
+}
+
+func TestMemoryEmptyMetricsForOnePod(t *testing.T) {
+	tc := testCase{
+		replicas:              4,
```
Comment (jszczepkowski) [inline pkg/controller/podautoscaler/metrics/metrics_client_test.go]: set it to 3

**Comment by socaa** [comment]
PTAL

@jszczepkowski 
@piosz

**Comment by jszczepkowski** [comment]
LGTM

**Comment by jszczepkowski** [comment]
Part of #14108 & #12087.

**Pull request merged**